### ISSUE TITLE
Restricted user Error and warning messages missing.

### ISSUE THREAD

**Issue by mamathac** [Juniper/py-junos-eznc#874]
PyEZ prints generic login error message for Restricted user:
Traceback (most recent call last):
  File "/var/tmp/testcase24.py", line 11, in <module>
    dev.open()
  File "/usr/local/lib/python2.7/dist-packages/jnpr/junos/device.py", line 1302, in open
    raise cnx_err
jnpr.junos.exception.ConnectError: ConnectError(host: nms5-mx480-b, msg: Extra content at the end of the document, line 6, column 13 (line 6))

Netconf Behaviour:
For user restricted, netconf prints below message.
user@router:/var/tmp# ssh Testuser1@router -s netconf
The authenticity of host 'router (IP)' can't be established.
ECDSA key fingerprint is c0:26:6a:cd:8d:d3:40:56:b7:a1:5c:8a:7a:ab:18:e3.
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added router,ip' (ECDSA) to the list of known hosts.
Password:
warning: user "TestUser1" does not have access privileges.
 
error: Restricted user session.

 
user TestUser1 {
    uid 1234;
    class unauthorized;
    authentication {
        encrypted-password "$1$BDbPDp6l$Rc1wXgntOalZ4T68MfCsv1"; ## SECRET-DATA
    }
}
 

While running PYEZ script with this user previleges,  now it returns ConnectError without any meaningful message unlike before.

Earlier version of PYEZ (1.4.0.dev) with Junos printed these warning and error messages.

Now it simply prints generic Connect fail error.

**Comment by stacywsmith** [comment]
This is a Junos bug. The device returned plain text rather than valid NETCONF/XML.

The device I tried, an SRX300 running Junos 17.3R1-S1.6, returned both plain text and NETCONF/XML. That's still a Junos bug, but not quite as bad as just returning plain text.

```
user@host$ ssh foo@srx -p830 -s netconf
Password:
warning: user "foo" does not have access privileges.

error: Restricted user session.
<!-- No zombies were killed during the creation of this user interface -->
<rpc-reply>
<rpc-error>
<error-severity>warning</error-severity>
<error-message>
user "foo" does not have access privileges.
</error-message>
</rpc-error>
</rpc-reply>
<rpc-reply>
<rpc-error>
<error-type>protocol</error-type>
<error-tag>operation-failed</error-tag>
<error-severity>error</error-severity>
<error-message>
Restricted user session.
</error-message>
</rpc-error>
</rpc-reply>
<!-- user nobody, class (unknown) -->
<hello xmlns="urn:ietf:params:xml:ns:netconf:base:1.0">
  <capabilities>
    <capability>urn:ietf:params:netconf:base:1.0</capability>
    <capability>urn:ietf:params:netconf:capability:candidate:1.0</capability>
    <capability>urn:ietf:params:netconf:capability:confirmed-commit:1.0</capability>
    <capability>urn:ietf:params:netconf:capability:validate:1.0</capability>
    <capability>urn:ietf:params:netconf:capability:url:1.0?scheme=http,ftp,file</capability>
    <capability>urn:ietf:params:xml:ns:netconf:base:1.0</capability>
    <capability>urn:ietf:params:xml:ns:netconf:capability:candidate:1.0</capability>
    <capability>urn:ietf:params:xml:ns:netconf:capability:confirmed-commit:1.0</capability>
    <capability>urn:ietf:params:xml:ns:netconf:capability:validate:1.0</capability>
    <capability>urn:ietf:params:xml:ns:netconf:capability:url:1.0?protocol=http,ftp,file</capability>
    <capability>urn:ietf:params:xml:ns:yang:ietf-netconf-monitoring</capability>
    <capability>http://xml.juniper.net/netconf/junos/1.0</capability>
    <capability>http://xml.juniper.net/dmi/system/1.0</capability>
  </capabilities>
  <session-id>78473</session-id>
</hello>
]]>]]>
<rpc><close-session/></rpc>
<rpc-reply xmlns="urn:ietf:params:xml:ns:netconf:base:1.0" xmlns:junos="http://xml.juniper.net/junos/17.3R1/junos">
<ok/>
</rpc-reply>
]]>]]>
<!-- session end at 2018-09-03 18:58:11 MDT -->
```

**Comment by vnitinv** [comment]
@mamathac Can you check the behaviour change in Junos version?

**Comment by mamathac** [comment]
Info provided to @vnitinv

**Comment by vnitinv** [comment]
@stacywsmith @mamathac We used to handle this at https://github.com/ncclient/ncclient/blob/master/ncclient/devices/junos.py#L52
looking into as in why things are broken on test server now.

**Comment by mamathac** [comment]
Working NCClient version **ncclient>=0.5.3 in /usr/local/lib/python2.7/dist-packages (from junos-eznc)**

**Comment by vnitinv** [comment]
@mamathac  we have sent pull request to ncclient https://github.com/ncclient/ncclient/pull/283

**Comment by vnitinv** [comment]
@mamathac the pull request to ncclient was merged. Hence please verify with latest ncclient.

**Comment by rsmekala** [comment]
Fixed via https://github.com/ncclient/ncclient/pull/283. Released via `ncclient-0.6.3`

Closing this issue. Please reopen if required ✌️
